src/main/java/seedu/address/logic/commands/EditStudentCommand.java

```diff
@@ -51,58 +52,62 @@
             throw new CommandException(MESSAGE_DUPLICATE_STUDENT);
         }
 
-        model.setStudent(studentToEdit, editedStudent);
-        model.updateFilteredStudentList(PREDICATE_SHOW_ALL_STUDENTS);
-        return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, editedStudent));
+        model.setTuition(tuitionToEdit, editedTuition);
+        model.updateFilteredTuitionList(PREDICATE_SHOW_ALL_STUDENTS);
+        return new CommandResult(String.format(MESSAGE_EDIT_STUDENT_SUCCESS, editedTuition));
     }
 
     /**
      * Creates and returns a {@code Student} with the details of {@code studentToEdit}
      * edited with {@code editStudentDescriptor}.
      */
-    private static Student createEditedStudent(Student studentToEdit, EditStudentDescriptor editStudentDescriptor) {
-        assert studentToEdit != null;
-
-        Name updatedName = editStudentDescriptor.getName().orElse(studentToEdit.getName());
-        Phone updatedPhone = editStudentDescriptor.getPhone().orElse(studentToEdit.getPhone());
-        Email updatedEmail = editStudentDescriptor.getEmail().orElse(studentToEdit.getEmail());
-        Address updatedAddress = editStudentDescriptor.getAddress().orElse(studentToEdit.getAddress());
-        String updatedStudyLevel = editStudentDescriptor.getStudyLevel().orElse(studentToEdit.getStudyLevel());
-        Phone updatedGuardianPhone = editStudentDescriptor.getGuardianPhone().orElse(studentToEdit.getGuardianPhone());
-        String updatedRelationship = editStudentDescriptor.getRelationship().orElse(studentToEdit.getRelationship());
-
-        return new Student(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedStudyLevel,
-            updatedGuardianPhone, updatedRelationship);
+    private static Tuition createEditedTuition(Tuition tuitionToEdit, EditTuitionDescriptor editTuitionDescriptor) {
+        assert tuitionToEdit != null;
+
+        Name updatedName = editTuitionDescriptor.getName().orElse(tuitionToEdit.getName());
+        Phone updatedPhone = editTuitionDescriptor.getPhone().orElse(tuitionToEdit.getPhone());
+        Email updatedEmail = editTuitionDescriptor.getEmail().orElse(tuitionToEdit.getEmail());
+        Address updatedAddress = editTuitionDescriptor.getAddress().orElse(tuitionToEdit.getAddress());
+        String updatedStudyLevel = editTuitionDescriptor.getStudyLevel()
+                .orElse(tuitionToEdit.getStudyLevel());
+        Phone updatedGuardianPhone = editTuitionDescriptor.getGuardianPhone()
+                .orElse(tuitionToEdit.getGuardianPhone());
+        String updatedRelationship = editTuitionDescriptor.getRelationship()
+                .orElse(tuitionToEdit.getRelationship());
+
+        Student student = new Student(updatedName, updatedPhone, updatedEmail, updatedAddress, updatedStudyLevel,
+                updatedGuardianPhone, updatedRelationship);
+        return new Tuition(student);
```

Comment:
Instead of returning a new Tuition object when we call edit_student, what about returning a student and replacing the student in the Tuition object? 
Because the command name is EditStudentCommand which does not really make sense to return a new Tuition object.